gateway/src/apicast/policy/mtls/mtls.lua

```diff
@@ -0,0 +1,41 @@
+local policy = require('apicast.policy')
+local _M = policy.new('mtls')
+local X509 = require('resty.openssl.x509')
+local errors = require('apicast.errors')
+local re_split = require('ngx.re').split
+
+local mt = {
+  __index = _M
+}
+
+local function encode(bytes)
+  local encoded = ngx.encode_base64(bytes)
+  return re_split(encoded, '=')[1]
+end
+
+local function check_certificate(cert, cnf)
+  if not (cert and cnf) then return false end
+
+  if encode(cert:digest('SHA256')) == cnf['x5t#S256'] then return true end

```

Comment:
what about if we move `x5t#S256` to a constant and we add a comment with the link to the description, it'll be helpful for the future. 

https://tools.ietf.org/html/draft-ietf-jose-json-web-signature-41#section-4.1.8

Reply:
There is no need to use it as a constant. The memory usage is going to be the same. And it is used only on one place anyway.

Reply:
Changed.